R/yaml.R

```diff
@@ -71,6 +73,15 @@ yaml_load = function(
   res
 }
 
+
+guess_indent_length = function(x) {
+  x = nchar(x)
+  y = min(setdiff(x, 0))
+  if(!all((x %% y) == 0))
+    stop('Irregular indentation in YAML header')
+  x / y
+}
```

Comment:
Indent always means the next level, even if not indenting deep enough:

```suggestion
indent_level = function(x) {
  N = nchar(x); n = N[N > 0]
  if (length(n) == 0) N else ceiling(N / min(n))
}
```

Reply:
I couldn't get to documentation at work (some weird router setting that blocks github.io and other websites).

https://yaml.org/spec/1.2.2/#61-indentation-spaces

The indentation rules are quite complex and depend only on the current parent.

So as long as tests are passing... hopefully no one will mix spaces with tabs, which are not even supported by YAML spec.